tests/e2e/specs/shopper/mini-cart.test.js

```diff
@@ -94,4 +123,89 @@ describe( 'Shopper → Mini Cart', () => {
 			);
 		} );
 	} );
+
+	describe( 'Filled mini cart', () => {
+		beforeAll( async () => {
+			await shopper.emptyCart();
+		} );
+
+		afterEach( async () => {
+			await shopper.emptyCart();
+		} );
+
+		it( 'The Mini Cart title shows correct amount', async () => {
+			await page.click(
+				'.wc-block-grid__product:first-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (1 item)',
+			} );
+
+			await page.mouse.click( 50, 200 );
+
+			await page.click(
+				'.wc-block-grid__product:last-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (2 items)',
+			} );
+		} );
+
+		it( 'The Mini Cart products table show added products', async () => {
+			const products = await page.$$(
+				'.wc-block-all-products .wc-block-grid__product'
+			);
+			if ( products.length === 0 ) {
+				return;
+			}
+
+			// Get a random product to ensure the test isn't false positive.
+			const product =
+				products[ Math.floor( Math.random() * products.length ) ];
+			const productTitle = await page.evaluate( ( el ) => {
+				return el.querySelector( '.wc-block-components-product-name' )
+					.textContent;
+			}, product );
```

Comment:
Could we use ElementHandle getProperty here -> `product.getProperty( 'textContent' )` for clarity?

Reply:
Ah same here... :) 

Would pattern like this make it clearer? 
```
const productTitleEl = await product.$( '.wc-block-components-product-name' );
const productTitle = await productTitleEl.getProperty( 'textContent' ); 
```

I'm pondering usage of evaluate vs more declarative use of the page/frame/elementHandle APIs from perspective of readability

Reply:
or in case of clicking the cart 

```
const productAddToCart = await product.$( '.add_to_cart_button' );
await productAddToCart.click();
```

Reply:
Applied your suggestion! Thanks!